File: frontend/src/model/Recipe.ts
import {Ingredient} from "./Ingredient";


export type Recipe = {
    id?: string,
    name: string,
    mealType: MealType | string,
    source: string,
    image: string,
    ingredients?: Ingredient[],
    prepTime: string,
    preparation: string,
    portions: number,
    favorite: boolean,
    dishTypeCategory?: DishTypeCategory | string,
    recipeCategory?: RecipeCategory | string,
    menuCategory?: MenuCategory | string,
    garnish?: string
}
export type NewRecipe = {

    name: string,
    source: string,
    image: string,
    prepTime: string,
    preparation: string,
    portions: number
    favorite: boolean
    garnish: string
}

export enum MealType {
    BREAKFAST = "BREAKFAST",
    LUNCH = "LUNCH",
    DINNER = "DINNER",

}

export enum DishTypeCategory {
    VEGGIE = "VEGGIE",
    MEAT = "MEAT",
    PASTA = "PASTA",
    FISH = "FISH",
    NOTHING_TODAY = "NOTHING_TODAY",
}

export enum RecipeCategory {
    SALAD = "SALAD",
    APPETIZER = "APPETIZER",
    SOUP = "SOUP",
    LOW_CARB = "LOW_CARB",
    HIGH_PROTEIN = "HIGH_PROTEIN",
}
export enum MenuCategory {
    ENTREE= "ENTREE",
    MAIN_COURSE ="MAIN_COURSE",
    DESSERT="DESSERT",
    SNACK="SNACK",
}

File: backend/src/main/java/com/github/licetrosales/backend/model/Recipe.java
package com.github.licetrosales.backend.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.With;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Collections;
import java.util.List;
@JsonIgnoreProperties(ignoreUnknown = true)
@With
@Document("recipes")
public record Recipe(
        @Id
        String id,
        String name,
        MealType mealType,
        String source,
        String image,
        List<Ingredient> ingredients,
        String prepTime,
        String preparation,
        int portions,
        Boolean favorite,
        DishTypeCategory dishTypeCategory,
        RecipeCategory recipeCategory,
        MenuCategory menuCategory,
        String garnish
) {
}
